src/simulation/in_memory_simulation_store.jl

```diff
@@ -172,7 +172,7 @@ function read_result(
     key::OptimizationContainerKey,
     index::EmulationModelIndexType,
 )
-    return read_results(get_em_data(store), key, index)
+    return read_results(get_em_data(store), key, index=index)
```

Comment:
This is unrelated to the cache issue, but was clearly wrong. If you fixed this in a different way in your branch such that this will cause a merge conflict, I can remove it.

Reply:
This is ok, I haven't run into this issue.